pygrametl/__init__.py

```diff
@@ -444,23 +444,38 @@ def now(ignoredtargetconn=None, ignoredrow=None, ignorednamemapping=None):
 
 
 def ymdparser(ymdstr):
-    """Convert a string of the form 'yyyy-MM-dd' to a datetime.date.
+    """Convert input of the form 'yyyy-MM-dd' to a datetime.date.
 
        If the input is None, the return value is also None.
+       If the input is a datetime.date, it is returned.
+       If the input is a datetime.datetime, its date is returned.
+       Else the input is cast to str and quotes are stripped before it is
+       split into the different parts needed to create a datetime.date.
     """
     if ymdstr is None:
         return None
+    elif isinstance(ymdstr, date):
+        return ymdstr
+    elif isinstance(ymdstr, datetime):
+        return ymdstr.date()
+    ymdstr = str(ymdstr).strip("'\"")
     (year, month, day) = ymdstr.split('-')
     return date(int(year), int(month), int(day))
 
 
 def ymdhmsparser(ymdhmsstr):
-    """Convert a string 'yyyy-MM-dd HH:mm:ss' to a datetime.datetime.
+    """Convert input of the form 'yyyy-MM-dd HH:mm:ss' to a datetime.datetime.
```

Comment:
_"input of the form 'yyyy-MM-dd HH:mm:ss'"_ implies that the input must be string, maybe we could write something like _"Convert a string of the form 'yyyy-MM-dd HH:mm:ss' or a datetime.datetime to a datetime.datetime."_ Although, that still does not mention that `None` is an acceptable input.

Reply:
Same reasoning as above. I'm not sure what would be the best description

Reply:
See this [conversation](https://github.com/chrthomsen/pygrametl/pull/54#discussion_r1032128638).

Reply:
Changed to a more precise description